hedgehog/src/Hedgehog/Internal/Runner.hs

```diff
@@ -174,14 +191,28 @@ checkReport cfg size0 seed0 test0 updateUI =
         -- size has reached limit, reset to 0
         loop tests discards 0 seed coverage0
 
+      else if failureVerified tests coverage0 then
+        -- tests have been verified to not reach coverage for at least one label
+        pure . Report tests discards coverage0 . Failed $
+          mkFailure
+            size
+            seed
+            0
+            (Just coverage0)
+            Nothing
+            "Test coverage cannot be reached, aborted"
+            Nothing
+            []
+
       else if tests >= fromIntegral (propertyTestLimit cfg) then
         -- we've hit the test limit
```

Comment:
One thing to note here is that just as we can stop early if we'll never reach the desired coverage - we *can* also stop early if we've already reached it!

I guess that is sort of a design decision - should we stop if we've already reached the desired coverage?

Reply:
Actually - turned out that there was a bug hiding if we didn't do this. Fixed it - I'll give this a rest now.